### PR TITLE
[Fix] `newline-after-import`: respect decorator annotations

### PR DESCRIPTION
**Pull request by lilling**
hey! found the bug. here in my pr enjoy!

how do I link it with #1784 ?

### CODE REVIEW

**Comment by lilling** [comment]
this is my fr pr ever to Github
I'll be glad if you guys can help me understand the failing checks

File: tests/src/rules/newline-after-import.js [hunk 1]
```diff
@@ -429,5 +429,16 @@ ruleTester.run('newline-after-import', require('rules/newline-after-import'), {
...
       parser: require.resolve('babel-eslint'),
     },
+    {
+      code: `// issue 1784\nimport { map } from 'rxjs/operators';\n@Component({})\nexport class Test {}`,
+      output: `// issue 1784\nimport { map } from 'rxjs/operators';\n\n@Component({})\nexport class Test {}`,
+      errors: [ {
+        line: 2,
+        column: 1,
+        message: IMPORT_ERROR_MESSAGE,
+      } ],
```
Comment (ljharb) [inline tests/src/rules/newline-after-import.js:439]:
```suggestion
      errors: [
        {
          line: 2,
          column: 1,
          message: IMPORT_ERROR_MESSAGE,
        },
      ],
```

File: tests/src/rules/newline-after-import.js [hunk 2]
```diff
@@ -429,5 +429,16 @@ ruleTester.run('newline-after-import', require('rules/newline-after-import'), {
       parserOptions: { sourceType: 'module' },
       parser: require.resolve('babel-eslint'),
     },
+    {
+      code: `// issue 1784\nimport { map } from 'rxjs/operators';\n@Component({})\nexport class Test {}`,
+      output: `// issue 1784\nimport { map } from 'rxjs/operators';\n\n@Component({})\nexport class Test {}`,
```
Comment (ljharb) [inline tests/src/rules/newline-after-import.js:434]:
```suggestion
      code: `
        // issue 1784
        import { map } from 'rxjs/operators';
        @Component({})
        export class Test {}
      `,
      output: `
        // issue 1784
        import { map } from 'rxjs/operators';
        
        @Component({})
        export class Test {}
      `,
```

**Review by ljharb — CHANGES_REQUESTED** [review]
Congrats! To link the PR with an issue, edit either the OP of the PR, and/or the commit message, to contain a phrase like `Fixes #1784` (or "closes", "resolves", etc - there's a bunch).

The appveyor WSL failures can be ignored, they always fail.

I'm not sure why the webpack resolver tests are failing; I'll look into it.

**Comment by ljharb** [comment]
Looks like today's npm 7 promotion caused some tests to break on master. I'll fix those, and then rebase your PR on top of it.

**Comment by lilling** [comment]
@ljharb do you have any updates?

**Comment by ljharb** [comment]
@lilling no, I’m afraid not :-/ it’s being tracked in #1986, and you don’t have to do anything else - once that’s done, I’ll rebase this, and we can get it in :-)

**Review by ljharb — APPROVED** [review]

**Comment by ljharb** [comment]
@lilling i've fixed master, so it looks like these are real failures. Please mark this PR as ready for review once you've figured them out :-)

**Comment by ljharb** [comment]
I'm seeing a few flaky failures in other PRs, so it might not actually be your branch. Stand by :-)

**Comment by lilling** [comment]
@ljharb Any updates?

**Comment by minwe** [comment]
```
ESLint: 7.26.0

TypeError: Cannot read property 'type' of null
Occurred while linting /src/components/Range/index.ts:3
    at isExportNameClass (/node_modules/_eslint-plugin-import@2.23.0@eslint-plugin-import/lib/rules/newline-after-import.js:54:69)
```

**Comment by ljharb** [comment]
@minwe in general, please file new issues for things like that. This one already is filed as #2056 and #2063.

**Pull request merged**